### ISSUE TITLE
Bug - DataList - additional actions cause onSelectDataListItem to fire via keyboard

### ISSUE THREAD

**Issue by thatblindgeye** [patternfly/patternfly-react#11390]
**Describe the problem**
In the [Clickable rows data list example](https://www.patternfly.org/components/data-list#clickable-rows), using keyboard to toggle the kebab of either data list item causes the data list item to become selected without opening the kebab.

**How do you reproduce the problem?**
Go to above example and try using Space or Enter when the kebab toggle has focus. Notice that the menu does not open and instead the row gets selected.

**Expected behavior**
The kebab/action should be triggered without triggering the selection update of the data list item.

**Is this issue blocking you?**
List the workaround if there is one.

**Screenshots**
If applicable, add screenshots to help explain the issue.

**What is your environment?**

**What is your product and what release date are you targeting?**

**Any other information?**
